src/frontend/src/handler/create_mv.rs

```diff
@@ -232,19 +220,23 @@ It only indicates the physical clustering of the data, which may improve the per
 "#.to_string());
         }
 
-        let (plan, table) = gen_create_mv_plan_bound(
-            &session,
-            context.into(),
-            query,
-            dependent_relations,
-            name,
-            columns,
-            emit_mode,
-        )?;
+        let (plan, table) =
+            gen_create_mv_plan_bound(&session, context.into(), query, name, columns, emit_mode)?;
+
+        let dependencies =
+            RelationCollectorVisitor::collect_with(dependent_relations, plan.clone())
```

Comment:
So for UDF, we record as long as we encounter one in the binder. But for relations, we record by visiting the _optimized_ plan. Is it possible to make this behavior consistent? Personally I think we should handle it in binder (i.e., regardless of optimization), as it can be confusing if a dropping a dependent relation is accepted even if it may invalidate the definition of the downstream job.

Reply:
> Personally I think we should handle it in binder (i.e., regardless of optimization)

Yes, I agree. I didn't change it because I want to mainly focus on UDF in this PR.

Reply:
added a todo comment